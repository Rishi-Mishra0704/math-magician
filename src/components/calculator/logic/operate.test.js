import operate from './operate';

test('addition', () => {
  const result = operate('2', '3', '+');
  expect(result).toMatchSnapshot();
});

test('subtraction', () => {
  const result = operate('5', '2', '-');
  expect(result).toMatchSnapshot();
});

test('multiplication', () => {
  const result = operate('7', '8', 'x');
  expect(result).toMatchSnapshot();
});

test('division', () => {
  const result = operate('10', '2', 'รท');
  expect(result).toMatchSnapshot();
});

test('division by zero', () => {
  const result = operate('5', '0', 'รท');
  expect(result).toMatchSnapshot();
});

test('modulo', () => {
  const result = operate('10', '3', '%');
  expect(result).toMatchSnapshot();
});

test('modulo with zero', () => {
  const result = operate('10', '0', '%');
  expect(result).toMatchSnapshot();
});

test('unknown operation', () => {
  expect(() => operate('1', '2', '&')).toThrowErrorMatchingSnapshot();
});
