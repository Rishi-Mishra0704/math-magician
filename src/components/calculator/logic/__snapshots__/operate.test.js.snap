// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addition 1`] = `"5"`;

exports[`division 1`] = `"5"`;

exports[`division by zero 1`] = `"Can't divide by 0."`;

exports[`modulo 1`] = `"1"`;

exports[`modulo with zero 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`multiplication 1`] = `"56"`;

exports[`subtraction 1`] = `"3"`;

exports[`unknown operation 1`] = `"Unknown operation '&'"`;
